src/codegen.cpp

```diff
@@ -8748,12 +8728,20 @@ static void gen_c_object(CodeGen *g, Buf *self_exe_path, CFile *c_file) {
             exit(1);
         }
 
-        // add the files depended on to the cache system
-        if ((err = cache_add_dep_file(cache_hash, out_dep_path, true))) {
-            fprintf(stderr, "Failed to add C source dependencies to cache: %s\n", err_str(err));
-            exit(1);
+        bool dep_file_exists;
+        if ((err = os_file_exists(out_dep_path, &dep_file_exists)))
```

Comment:
One last improvement: cache_add_dep_file should correctly return ErrorFileNotFound, and so this should be handling that case rather than a separate check.

Reply:
`cache_add_dep_file` is run with `verbose` set to `true`, that makes it print out a few useful errors in case the file fails to parse but also makes it spit out an error in case the file is not found.

Reply:
Hmm. Good point. I'd rather set that to false and lose the verbosity than do the extra file system check. Or, make ErrorFileNotFound special in the implementation of the function so that verbose does not apply to it.

Reply:
Done, I went for the second option.